frontend/src/metabase/components/Schedule/Schedule.tsx

```diff
@@ -161,99 +180,139 @@ const ScheduleBody = ({
-      </>
+      </TwoColumns>
     );
   } else if (scheduleType === "monthly") {
     // e.g. "Send monthly on the 15th at 12:00pm"
     if (schedule.schedule_frame === "mid") {
       return (
-        <>
+        <TwoColumns>
           {c(
             "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is the noun '15th' (as in 'the 15th of the month'), {3} is a time like '12:00pm'",
           ).jt`${verb} ${(<SelectFrequency {...frequencyProps} />)} on the ${(
             <SelectFrame {...frameProps} />
-          )} at ${(<SelectTime {...timeProps} />)} `}
-          <TimeDetails {...timeDetailsProps} />
-        </>
+          )} at ${(<SelectTime {...timeProps} />)}`}
+        </TwoColumns>
       );
     } else {
       // e.g. "Send monthly on the first Tuesday at 12:00pm"
       return (
-        <>
+        <TwoColumns>
           {c(
             "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is an adjective like 'first', {3} is a day like 'Tuesday', {4} is a time like '12:00pm'",
           ).jt`${verb} ${(<SelectFrequency {...frequencyProps} />)} on the ${(
             <SelectFrame {...frameProps} />
           )} ${(<SelectWeekdayOfMonth {...weekdayOfMonthProps} />)} at ${(
             <SelectTime {...timeProps} />
           )}`}
-        </>
+        </TwoColumns>
       );
     }
   } else {
     return null;
   }
 };
 
+/** Arrange Schedule components into two columns, with the Selects on the right. */
+const TwoColumns = ({ children }: { children: ReactNode }) => {
+  const kids = Children.toArray(children).filter(
+    child => !(typeof child === "string" && !child.trim()),
+  );
+  const result: ReactNode[] = [];
+  const addBlank = () => result.push(<Box></Box>);
+  for (let c = 0; c < kids.length; c++) {
+    const curr = kids[c];
+    const next = kids[c + 1];
+    const isLastItemString = c === kids.length - 1 && typeof curr === "string";
+    if (isLastItemString) {
+      addBlank();
+      result.push(<Box mt="-.5rem">{curr}</Box>);
+    } else {
+      const isFirstItemString = c === 0 && typeof curr !== "string";
+      if (isFirstItemString) {
+        addBlank();
+      }
+      if (typeof curr === "string") {
+        const wrappedCurr = <Box style={{ textAlign: "end" }}>{curr}</Box>;
+        result.push(wrappedCurr);
+      } else {
+        result.push(curr);
+      }
+    }
+    // Insert blank nodes between adjacent Selects unless they can fit on one line
+    if (isValidElement(curr) && isValidElement(next)) {
+      const canSelectsProbablyFitOnOneLine =
+        curr.props.longestLabel.length + next.props.longestLabel.length < 24;
+      if (canSelectsProbablyFitOnOneLine) {
+        result[result.length - 1] = (
+          <Group spacing="xs">
+            {result[result.length - 1]}
+            {next}
+          </Group>
+        );
+        c++;
+      } else {
+        addBlank();
+      }
+    }
+  }
+  return <>{result}</>;
+};
```

Comment:
I know this is kind of evil, but it seems necessary in order to restructure the translated JSX so that the Selects appear on the right

Note that a translator can re-order the placeholders, for example by putting the verb at the end, as in German. This function inserts blank `Box`es so that everything flows nicely in the grid (defined via gridTemplateColumns above)

Reply:
Talked over a huddle, I think we came up with a strategy to parse a string and inject components rather than trying to parse / remap react children.